### ISSUE TITLE
torchtext (nightly) has removed torchtext.legacy, causing pytorch-lightning to fail

### ISSUE THREAD

**Issue by tridao** [PyTorchLightning/pytorch-lightning#11662]
## 🐛 Bug
The nightly version of torchtext has removed torchtext.legacy, so pytorch-lightning fails.
```
File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/utilities/apply_func.py", line 30, in <module>
    from torchtext.legacy.data import Batch
ModuleNotFoundError: No module named 'torchtext.legacy'
```

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): 1.5.9
- PyTorch Version (e.g., 1.10): 1.11 nightly
- Python version (e.g., 3.9): 3.8
- OS (e.g., Linux): Ubuntu 20.04

**Comment by akihironitta** [comment]
@tridao What's the output of `python -c "import torchtext; print(torchtext.__version__)"` in your env?

**Comment by tridao** [comment]
It's `0.12.0a0`.
Btw I'm using the [nvcr.io/nvidia/pytorch:22.01-py3](https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel_22-01.html#rel_22-01) docker image.

**Comment by akihironitta** [comment]
Closing this as it's been addressed already in #10597.
